File: RoRClient/Model/Models/InteractiveGameObject.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RoRClient.Model.Models
{
    public abstract class InteractiveGameObject : ModelBase, INotifyPropertyChanged
    {
        //Abstrakte Klasse für alle Objekte, mit denen interagiert werden kann

        public event PropertyChangedEventHandler PropertyChanged;
        public virtual void OnPropertyChanged(string propertyName)
        {
            if (PropertyChanged != null)
            {
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }

        private Square square;
        public Square Square
        {
            get
            {
                return square;
            }
            set
            {
                if(square != value)
                {
                    square = value;
                    OnPropertyChanged("Square");            
                }
            }
        }

        public InteractiveGameObject(Square square)
        {
            this.square = square;
        }
    }
}

File: RoRServer/src/main/java/models/base/InteractiveGameObject.java
package models.base;

/**
 * Abstrakte Klasse fuer alle Objekte, mit denen interagiert werden kann.
 * Sind Observer und Observable
 */
public abstract class InteractiveGameObject extends InterActiveGameModel {
	protected final String className;
    
	public InteractiveGameObject() {
		super();
		this.className = getClass().getName();
	}
	
	public String getClassName() {
		return className;
	}
}
